tools/testing/target_determination/gen_artifact.py

```diff
@@ -1,10 +1,12 @@
 import json
+import os
 import pathlib
 from typing import Any, List
 
 REPO_ROOT = pathlib.Path(__file__).resolve().parent.parent.parent.parent
 
 
 def gen_ci_artifact(included: List[Any], excluded: List[Any]) -> None:
-    with open(REPO_ROOT / "test/test-reports/td_exclusions.json", "w") as f:
+    file_name = f"td_exclusions-{os.urandom(10).hex()}.json"
+    with open(REPO_ROOT / f"test/test-reports/{file_name}.json", "w") as f:
```

Comment:
Nit (also, previous implementation will add .json extension twice
```suggestion
    with open(REPO_ROOT / "test" / " test-reports/ file_name, "w") as f:
```